homeassistant/components/vegehub/sensor.py

```diff
@@ -0,0 +1,210 @@
+        elif data_type == OPTION_DATA_TYPE_CHOICES[1]:
+            self._unit_of_measurement = PERCENTAGE
+            self._attr_device_class = SensorDeviceClass.MOISTURE
+        elif data_type == OPTION_DATA_TYPE_CHOICES[2]:
+            self._unit_of_measurement = UnitOfTemperature.CELSIUS
+            self._attr_device_class = SensorDeviceClass.TEMPERATURE
+        else:
+            self._unit_of_measurement = UnitOfElectricPotential.VOLT
+            self._attr_device_class = SensorDeviceClass.VOLTAGE
+
+        self._mac_address: str = mac_address
+        self._slot: int = slot
+        self._attr_unique_id: str = new_id
+        self.entity_id: str = "sensor." + new_id
+        self._ip_addr: str = ip_addr
+        self._dev_name: str = dev_name
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return self._attr_name
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the state of the sensor."""
+        if (
+            self._data_type == OPTION_DATA_TYPE_CHOICES[1] and self._attr_native_value
+        ):  # Percentage
+            return VH400_transform(self._attr_native_value)
+        if (
+            self._data_type == OPTION_DATA_TYPE_CHOICES[2] and self._attr_native_value
+        ):  # Temperature C
+            if isinstance(self._attr_native_value, (int, str, float)):
+                return (41.6700 * float(self._attr_native_value)) - 40.0000
+            return None
+
+        if isinstance(self._attr_native_value, (int, str, float)):
+            return float(self._attr_native_value)
+        return None
+
+    @property
+    def suggested_unit_of_measurement(self) -> str:
+        """Return the unit of measurement of this entity."""
+        return self._unit_of_measurement
+
+    @property
+    def native_unit_of_measurement(self) -> str:
+        """Return the unit of measurement of this entity."""
+        return self._unit_of_measurement
+
+    @property
+    def device_class(self) -> SensorDeviceClass | None:
+        """Return the class of this sensor."""
+        return self._attr_device_class
+
+    @property
+    def suggested_display_precision(self) -> int:
+        """Return the suggested display precision of this sensor."""
+        return 2
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this entity."""
+        return self._attr_unique_id
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device info."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self._mac_address)},
+            name=self._dev_name,
+            manufacturer=MANUFACTURER,
+            model=MODEL,
+        )
+
+    async def async_update_sensor(self, value):
+        """Update the sensor state with the latest value."""
+
+        self._attr_native_value = value
+        self.async_write_ha_state()
+
+
+def VH400_transform(x: Any) -> float:
```

Comment:
ideally this would be in the library. If it absolutely can't be move it to helpers.py

Reply:
Moved this to the library. Commit b65a678.